docs/how-to/gpu-enabled-mpi.rst

```diff
@@ -0,0 +1,260 @@
+***************************************************************************************************
+GPU-enabled MPI
+***************************************************************************************************
+
+The Message Passing Interface (`MPI <https://www.mpi-forum.org>`_) is a standard API for distributed
+and parallel application development that can scale to multi-node clusters. To facilitate the porting of
+applications to clusters with GPUs, ROCm enables various technologies. You can use these
+technologies add GPU pointers to MPI calls and enable ROCm-aware MPI libraries to deliver optimal
+performance for both intra-node and inter-node GPU-to-GPU communication.
+
+The AMD kernel driver exposes remote direct memory access (RDMA) through *PeerDirect* interfaces.
+This allows network interface cards (NICs) to directly read and write to RDMA-capable GPU device
+memory, resulting in high-speed direct memory access (DMA) transfers between GPU and NIC. These
+interfaces are used to optimize inter-node MPI message communication.
+
+The Open MPI project is an open source implementation of the MPI. It's developed and maintained by
+a consortium of academic, research, and industry partners. To compile Open MPI with ROCm support,
+refer to the following sections:
+
+* :ref:`open-mpi-ucx`
+* :ref:`open-mpi-libfabric`
+
+.. _open-mpi-ucx:
+
+ROCm-aware Open MPI on InfiniBand and RoCE networks using UCX
+================================================================
+
+The `Unified Communication Framework <https://www.openucx.org/documentation>`_ (UCX), is an
+open source, cross-platform framework designed to provide a common set of communication
+interfaces for various network programming models and interfaces. UCX uses ROCm technologies to
+implement various network operation primitives. UCX is the standard communication library for
+InfiniBand and RDMA over Converged Ethernet (RoCE) network interconnect. To optimize data
+transfer operations, many MPI libraries, including Open MPI, can leverage UCX internally.
+
+To set up UCX so that it can be used to compile Open MPI, refer to the following instructions.
+
+1. Set environment variables to install all software components in the same base directory. We use the
+home directory in our example, but you can specify a different location if you want.
+
+    .. code-block:: shell
+
+        export INSTALL_DIR=$HOME/ompi_for_gpu
+        export BUILD_DIR=/tmp/ompi_for_gpu_build
+        mkdir -p $BUILD_DIR
+
+2. Instal UCX. To view UCX and ROCm version compatibility, refer to the
+`communication libraries tables <https://rocm.docs.amd.com/en/latest/reference/gpu_libraries/communication.html>`_.
```

Comment:
@edgargabriel I think we need to change this link: do you mind verifying?

Reply:
the correct link should be : https://rocm.docs.amd.com/en/latest/release/3rd_party_support_matrix.html